tests/export_test.py

```diff
@@ -57,19 +58,36 @@ def tearDownModule():
   prev_xla_flags()
 
 ### Setup for testing lowering with effects
-class TestingOrderedEffect1(effects.Effect):
-  __str__ = lambda _: "TestingOrderedEffect1"
+@dataclasses.dataclass(frozen=True)
+class ForTestingOrderedEffect1(effects.Effect):
+  __str__ = lambda _: "ForTestingOrderedEffect1"
 
-class TestingOrderedEffect2(effects.Effect):
-  __str__ = lambda _: "TestingOrderedEffect2"
+@dataclasses.dataclass(frozen=True)
+class ForTestingOrderedEffect2(effects.Effect):
+  __str__ = lambda _: "ForTestingOrderedEffect2"
+
+@dataclasses.dataclass(frozen=True)
+class ForTestingUnorderedEffect1(effects.Effect):
+  __str__ = lambda _: "ForTestingUnorderedEffect1"
+
+
+class ForTestingOrderedEffect4NoNullary(effects.Effect):
```

Comment:
Is this effect class deliberately not a `dataclass`?

Reply:
This is how most Effect classes are defined in JAX, and I wanted to have a test of that form.